### ISSUE TITLE
[Bug] error 128 when creating NEW Wiki page

### ISSUE THREAD

**Issue by juliusdeblaaij** [go-gitea/gitea#13846]
- Gitea version (or commit ref): 1.12.6
- Git version: 2.23.2.windows.1
- Operating system: Windows 10 Home 64-bit
I executed an exe from from the releases page
I run gitea by executing the .exe, which opens and runs
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
- Log gist:
[gitea.log](https://gist.github.com/juliusdeblaaij/46f6d7e295b857c2dc0de1a4f3f1d7e8)
[macaron.log](https://gist.github.com/juliusdeblaaij/7e457dd4623e4bdf93c767cb16385c8b)

## Description
When trying to create a new Wiki page I get the following 500 error:
`exit status 128 - fatal: your current branch 'main' does not have any commits yet`
which happens in both repo's with 'main' as a branch name as well as one's with 'master'

**Comment by gimpf** [comment]
I also encounter this bug, and I think I can add something useful.

For evaluation only I just tried gitea 1.13 on Windows 10 with sqlite3 and git 2.29.2, running from command line with current user (not service).

When creating a wiki page, it always ends up with a 500.

Reason behind is that gitea wiki obviously wants to use branch "master", but when creating the page it defaults to git's configuration, where I recently set `git config --global init.defaultBranch main`.

As soon as I set the default branch back to `master`, gitea works as expected.
